### PR TITLE
added post-processing for local gemma

### PR DESCRIPTION
**Pull request by lkuligin**

### CODE REVIEW

File: libs/vertexai/langchain_google_vertexai/gemma.py [hunk 1]
```diff
@@ -53,6 +53,21 @@ def gemma_messages_to_prompt(history: List[BaseMessage]) -> str:
...
+    pos = response.rfind(pattern)
+    if pos == -1:
+        return response
+    else:
+        text = response[(pos + len(pattern)) :]
+        pos = text.find("<start_of_turn>user\n")
+        if pos > 0:
+            return text[:pos]
+        else:
+            return text
```
Comment (holtskinner) [inline libs/vertexai/langchain_google_vertexai/gemma.py:68]:
```suggestion
    text = response[(pos + len(pattern)) :]
    pos = text.find("<start_of_turn>user\n")
    if pos > 0:
        return text[:pos]
    return text
```

Comment (holtskinner) [reply libs/vertexai/langchain_google_vertexai/gemma.py:68]: else's not needed

File: libs/vertexai/langchain_google_vertexai/gemma.py [hunk 2]
```diff
@@ -139,10 +159,12 @@ async def _agenerate(
         """Top Level call"""
         request = self._get_params(**kwargs)
         request["prompt"] = gemma_messages_to_prompt(messages)
-        output = await self.async_client.predict(
+        text = await self.async_client.predict(
```
Comment (holtskinner) [inline libs/vertexai/langchain_google_vertexai/gemma.py:162]: maybe change this back to `output` to be explicit that it's the raw output.

**Review by holtskinner — APPROVED** [review]

File: libs/vertexai/langchain_google_vertexai/gemma.py (hunk 1 above)
Comment (lkuligin) [reply libs/vertexai/langchain_google_vertexai/gemma.py:68]: ha, true! fixed!

File: libs/vertexai/langchain_google_vertexai/gemma.py (hunk 2 above)
Comment (lkuligin) [reply libs/vertexai/langchain_google_vertexai/gemma.py]: fixed!

**Pull request merged**